### ISSUE TITLE
Add Instance Matching to Orders Mapper

### ISSUE THREAD

**Issue by bltravis** [FOLIO-FSE/folio_migration_tools#394]
Add support for matching existing instance records to POLs when creating new composite order records.

**Comment by branchedelac** [comment]
@bltravis @fontanka16 I was thinking about that new "rules" feature in the record mapping file we looked at last week.

Could that be used for getting UUIDs of existing records? For example:

```
{
    "folio_field": "instanceId",
    "legacy_field": "",
    "value": "",
    "description": "",
    "fallback_legacy_field": "",
    "rules": {
        "fetchUuidfromIdMap": {
            "id_map": "iteration1_instances_id_map.json",
            "legacy_match_field": "bibId",
        }
    }
}
```

```
{
    "folio_field": "vendor",
    "legacy_field": "",
    "value": "",
    "description": "",
    "fallback_legacy_field": "",
    "rules": {
        "fetchUuidfromFOLIO": {
            "endpoint": "organizations/organizations",
            "legacy_match_field": "vendor",
            "folio_match_field": "name"
        }
    }
}
```

**Comment by branchedelac** [comment]
It struck me that instructors in Courses are another case where, like with instances/titles in Orders, you can chose either to add a string or to link the UUID of an existing object in FOLIO.

**Comment by fontanka16** [comment]
Fixed with commit 682bd1f52958040ce4cb26d986817c4d154cafcb
